src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneIndexingService.cs

```diff
@@ -106,57 +136,78 @@ public async Task ProcessContentItemsAsync()
+                    {
+                        if (indexSettings.IndexedContentTypes.Length == 0)
                         {
-                            var contentItem = await contentManager.GetAsync(task.ContentItemId);
+                            continue;
+                        }
 
-                            if (contentItem == null)
+                        foreach (var task in batch)
+                        {
+                            if (task.Type == IndexingTaskTypes.Update)
                             {
-                                continue;
-                            }
+                                var contentItem = await contentManager.GetAsync(task.ContentItemId);
 
-                            var context = new BuildIndexContext(new DocumentIndex(task.ContentItemId), contentItem, new string[] { contentItem.ContentType });
+                                if (contentItem == null)
+                                {
+                                    continue;
+                                }
 
-                            // Update the document from the index if its lastIndexId is smaller than the current task id. 
-                            await indexHandlers.InvokeAsync(x => x.BuildIndexAsync(context), Logger);
+                                var context = new BuildIndexContext(new DocumentIndex(task.ContentItemId), contentItem, new string[] { contentItem.ContentType });
 
-                            foreach (var index in allIndices)
-                            {
-                                if (index.Value < task.Id)
+                                // Update the document from the index if its lastIndexId is smaller than the current task id. 
+                                await indexHandlers.InvokeAsync(x => x.BuildIndexAsync(context), Logger);
+
+                                foreach (var index in allIndices)
                                 {
-                                    updatedDocumentsByIndex[index.Key].Add(context.DocumentIndex);
+                                    // Ignore if the content item content type is not indexed in this index
+                                    if (!indexSettingsList.Where(x => x.IndexName == index.Key).FirstOrDefault().IndexedContentTypes.Contains(contentItem.ContentType))
+                                    {
+                                        continue;
+                                    }
+
+                                    if (index.Value < task.Id)
+                                    {
+                                        updatedDocumentsByIndex[index.Key].Add(context.DocumentIndex);
+                                    }
                                 }
                             }
                         }
                     }
 
                     // Delete all the existing documents
-                    foreach (var index in allIndices)
+                    foreach (var index in updatedDocumentsByIndex)
```

Comment:
@sebastienros please see this method instead of doing a foreach on AllIndices which is not required here.